staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4604,8 +4604,10 @@ type NodeSpec struct {
 	// If specified, the node's taints.
 	// +optional
 	Taints []Taint `json:"taints,omitempty" protobuf:"bytes,5,opt,name=taints"`
-	// If specified, the source to get node configuration from
+
+	// Deprecated. If specified, the source to get node configuration from
 	// The DynamicKubeletConfig feature gate must be enabled for the Kubelet to use this field
+	// This field is deprecated as of 1.22: https://git.k8s.io/enhancements/keps/sig-node/281-dynamic-kubelet-configuration
```

Comment:
add a check in nodeStrategy#WarningsOnCreate and nodeStrategy#WarningsOnUpdate if this field is non-nil to return a warning:

```
spec.configSource: deprecated in v1.22+, support removal is planned in v1.X
```

Reply:
added, thank you for suggestion